### PR TITLE
chore: Update otel-collector to v0.89.0

### PR DESCRIPTION
**Pull request by karlkfi**
- Should fix the CVE in v0.87

### CODE REVIEW

File: manifests/templates/otel-collector.yaml [hunk 1]
```diff
@@ -93,7 +93,7 @@ spec:
     spec:
       containers:
       - name: otel-collector
-        image: gcr.io/config-management-release/otelcontribcol:v0.87.0-gke.2
```
Comment (sdowell) [inline manifests/templates/otel-collector.yaml:96]: Where is the `-gke.N` suffix?

Comment (nan-yu) [reply manifests/templates/otel-collector.yaml:96]: +1. The image should be pushed by the Louhi flow, which appends `-gke.N`.

Comment (karlkfi) [reply manifests/templates/otel-collector.yaml:96]: Does it need it? This is just the regular tag. https://gke-internal.googlesource.com/config-sync/open-telemetry/opentelemetry-collector-contrib/+/refs/tags/v0.89.0

It wasn't clear to me if there all get built or if i need to make a release branch to trigger a gke build.

Comment (karlkfi) [reply manifests/templates/otel-collector.yaml:96]: hmm. maybe these don't exist

Comment (sdowell) [reply manifests/templates/otel-collector.yaml:96]: It looks like that tag hasn't been imported yet. You don't need to make a release branch, it should be automated. It's possible something went wrong, I can take a closer look

Comment (sdowell) [reply manifests/templates/otel-collector.yaml:96]: Either way, do we want to upgrade minor versions? I think we should be deliberate about upgrading minor versions with a review of potential breaking changes. If it's just for CVE fixes, it's lower risk to create a new patch. Not to say we shouldn't have a cadence for upgrading minor versions

Comment (karlkfi) [reply manifests/templates/otel-collector.yaml:96]: the branch didn't exist. so i just made one... should i not have?

Comment (karlkfi) [reply manifests/templates/otel-collector.yaml:96]: the louhi build isn't triggered yet. should i hit it manually or is it just delayed for polling?

**Comment by sdowell** [comment]
replaced by https://github.com/GoogleContainerTools/kpt-config-sync/pull/1046

**Pull request closed without merging**